test/utils/HtmlFormatterSpec.scala

```diff
@@ -0,0 +1,34 @@
+/*
+ * Copyright 2020 HM Revenue & Customs
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package utils
+
+import org.scalatest.Matchers
+import uk.gov.hmrc.play.test.UnitSpec
+import utils.TaxsValidator._
+
+class HtmlFormatterSpec extends UnitSpec with Matchers {
+
+  "toHtmlNonBroken" should {
+    "replace spaces with non breaking spaces" in {
+      val stringBeforeReplace = "6 April 2018 to 5 April 2019"
+
+      val expectedString = "6&nbsp;April&nbsp;2018&nbsp;to&nbsp;5&nbsp;April&nbsp;2019"
```

Comment:
This pattern does not match the accessibility feedback
`6&nbsp;April&nbsp;2018 to 5&nbsp;April&nbsp;2019`
Non breaking spaces  only between the words in the date, not between every word.